src/seedu/addressbook/parser/Parser.java

```diff
@@ -226,5 +229,33 @@ private Command prepareFind(String args) {
         return new FindCommand(keywordSet);
     }
 
+    /**
+     * Parses arguments in the context of the sort command.
+     *
+     * @param args full command args string
+     * @return the prepared command
+     */
+    private Command prepareSort(String args) {
+        if (args.isEmpty()) {
+            return new SortCommand();
+        }
+        final Matcher matcher = KEYWORDS_ARGS_FORMAT.matcher(args.trim());
+        if (!matcher.matches()) {
+            return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                    SortCommand.MESSAGE_USAGE));
+        }
+
+        // keywords delimited by whitespace
+        final String[] keywords = matcher.group("keywords").split("\\s+");
+        final String keyword = keywords[0];
+
+        if (keyword.equalsIgnoreCase("desc") || keyword.equalsIgnoreCase("descending")) {
+            return new SortCommand(false);
+        } else if (keyword.equalsIgnoreCase("asc") || keyword.equalsIgnoreCase("ascending")) {
+            return new SortCommand();
+        }
```

Comment:
With respect to the previous comment, this should've just been SortCommand(true)
If the coded needed to be cleaner, a way would be to refactor out the true/false into its own variable, and create the object once only, so that we know what it means without the IDE autofill